colosseum/validators.py

```diff
@@ -103,3 +106,39 @@ def is_color(value):
 
 
 is_color.description = '<color>'
+
+
+_CSS_IDENTIFIER_RE = re.compile(r'^[a-zA-Z][a-zA-Z0-9\-\_]+$')
+
+
+def is_font_family(value):
+    """Validate that value is a valid font family."""
+    value = ' '.join(value.strip().split())
+    values = [v.strip() for v in value.split(',')]
+    checked_values = []
+    generic_family = ['serif', 'sans-serif', 'cursive', 'fantasy', 'monospace']
+    for val in values:
+        if (val.startswith('"') and val.endswith('"')
+                or val.startswith("'") and val.endswith("'")):
+            # TODO: Check that the font exists?
+            try:
+                ast.literal_eval(val)
+                checked_values.append(val)
+            except ValueError:
+                raise ValidationError
+        elif val in generic_family:
+            checked_values.append(val)
+        else:
+            # TODO: Check that the font exists?
+            if _CSS_IDENTIFIER_RE.match(val):
```

Comment:
Is there a CSS spec reference on this character restriction?

Reply:
https://www.w3.org/TR/CSS22/fonts.html#font-family-prop

>Font family names must either be given quoted as strings, or unquoted as a sequence of one or more identifiers. This means most punctuation characters and digits at the start of each token must be escaped in unquoted font family names.

>For example, the following declarations are invalid:

[On identifiers](https://www.w3.org/TR/CSS22/syndata.html#value-def-identifier)

Reply:
Ok - in which case, this (or a section reference) should be in the code as a docstring (probably around the regex declaration, rather than the usage). Essentially, I'm aiming for this to be as close to "literate code" as we can manage - any policy decision in code should be referencing the standard so that we can easily identify which decisions are arbitrary implementation decisions, and which are required by the standard.

The examples of invalid identifiers should also be included as test cases.